eta/core/datasets.py

```diff
@@ -738,32 +738,33 @@ def add_file(self, data_path, labels_path, new_data_filename=None,
                 data file already present in the dataset and
                 `error_on_duplicates` is True, or if `file_method` is not valid
         '''
-        if error_on_duplicates and self.has_data_with_name(data_path):
-            raise ValueError("Data file '%s' already present in dataset"
-                             % os.path.basename(data_path))
-
-        data_subdir = os.path.join(self.data_dir, self._DATA_SUBDIR)
         if new_data_filename is None:
             new_data_filename = os.path.basename(data_path)
 
-        labels_subdir = os.path.join(self.data_dir, self._LABELS_SUBDIR)
         if new_labels_filename is None:
             new_labels_filename = os.path.basename(labels_path)
 
+        if error_on_duplicates and self.has_data_with_name(new_data_filename):
+            raise ValueError("Data file '%s' already present in dataset"
+                             % new_data_filename)
+
         data_method, labels_method = self._parse_file_methods(file_method)
 
-        new_data_path = os.path.join(data_subdir, new_data_filename)
+        new_data_path = os.path.join(
+            self.data_dir, self._DATA_SUBDIR, new_data_filename)
         if data_path != new_data_path:
             data_method(data_path, new_data_path)
 
-        new_labels_path = os.path.join(labels_subdir, new_labels_filename)
+        new_labels_path = os.path.join(
+            self.data_dir, self._LABELS_SUBDIR, new_labels_filename)
         if labels_path != new_labels_path:
             labels_method(labels_path, new_labels_path)
-            # renames the 'filename' attribute within the json file to reflect
-            # the updated data_filename
-            data = load_json(new_labels_path)
-            data['filename'] = os.path.basename(new_data_path)
-            write_json(data, new_labels_path)
+
+        # Update the filename attribute in the labels JSON if necessary
+        if new_data_filename != os.path.basename(data_path):
+            labels_ = self._read_labels(new_labels_path)
```

Comment:
`labels_`? what is this convention?

Reply:
I wanted to use a variable name like `labels`, but was considered it was used elsewhere.  The underscore denotes a short-lived temp variable.

Reply:
this is nit, I know, but isn't our convention a prepended underscore, not appended?

Reply:
I've seen both in our code base.  Prepend means it's a private variable.